kubernetes-1.31.yaml

```diff
@@ -161,17 +156,27 @@ subpackages:
 
           install -d ${{targets.subpkgdir}}/var/log/kube-apiserver
 
-  - name: kubernetes-pause-3.9
-    description: ""
+  - name: kubernetes-pause-${{vars.kubernetes-version}}
     dependencies:
       provides:
-        - kubernetes-pause=3.9
+        - kubernetes-pause=${{vars.kubernetes-version}}
     pipeline:
       - working-directory: /home/build/build/pause
         runs: |
           mkdir -p ${{targets.subpkgdir}}/usr/bin/
-          CFLAGS="$CFLAGS -static -DVERSION=${{vars.pause-version}}-${{package.version}}"
-          gcc -o "${{targets.subpkgdir}}"/usr/bin/pause-3.9 linux/pause.c
+          CFLAGS="$CFLAGS -static -DVERSION=v$(grep '^TAG ?=' Makefile | awk '{print $3}')-${{package.version}}"
+          gcc ${CFLAGS} -o "${{targets.subpkgdir}}"/usr/bin/pause linux/pause.c
+    test:
+      pipeline:
+        - name: kubernetes pause version check
+          runs: /usr/bin/pause -V
+
+  - name: kubernetes-pause-compat-${{vars.kubernetes-version}}
```

Comment:
Where is this used?

Reply:
Apologies upfront because you're not going to like this answer. :( 

```bash
$ crane config registry.k8s.io/pause:3.10 | jq .config.Entrypoint
[
  "/pause"
]
```

Ahh, just having some fun before I go to bed, following is the detailed answer: 
```bash
$ kind create cluster
$ docker exec -it kind-control-plane bash
root@kind-control-plane:/# ctr --namespace=k8s.io container ls | grep pause
20807e1436af1b80ac01d970f82a34b86f53a7e3ef06d41cb26b5bd5e20d1250    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
3209afc8f95c83d3c21f385483eaf6f17769f2b51b27c6eb58b0ecb845212479    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
5dfb4e335ac6fe4747299255791bfa971135715fa3995f37e7c9845f8b1eeb7a    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
68bcace43965b022f1e4b8f207582ddd35b5f5c14298c8a11c42fa5bbfae7b4e    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
6bfa094639f1acb6887e4bfe26209b0274986f5fadae72b7ac716748e8d58406    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
6cf9d697c3c7635d1e28d3e787bd69c57dff2bd07e2f6a337667f519f9553316    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
e054de8075e5e368e1c3b0e2dc040fd2579ed711955ab69f7a03dca3090d0b99    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
e7ef02fefec6a2c698ac7adac1c26b4bc0baf46c06d0ec1dc0bc1a5d657a6bc4    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2
f49d08d1ed70f6e5472f581bb016795771a15932dc687b0edb0f22079ea377ff    registry.k8s.io/pause:3.10                                     io.containerd.runc.v2

root@kind-control-plane:/# ctr --namespace=k8s.io container info 20807e1436af1b80ac01d970f82a34b86f53a7e3ef06d41cb26b5bd5e20d1250 | jq .Spec.process.args
[
  "/pause"
]
```
the process (containerd in this case) at runtime calls the binary by absolute path.